src/main/java/org/mqttbee/mqtt/codec/encoder/mqtt3/Mqtt3PubRecEncoder.java

```diff
@@ -31,9 +31,8 @@
  */
 public class Mqtt3PubRecEncoder extends MqttMessageEncoderWithMessage<MqttPubRec> {
 
-    public static final Mqtt3PubRecEncoder INSTANCE = new Mqtt3PubRecEncoder();
     public static final MqttPubRecEncoderProvider PROVIDER =
-            new MqttPubRecEncoderProvider(() -> INSTANCE, Mqtt3PubRelEncoder.PROVIDER);
+            new MqttPubRecEncoderProvider(Mqtt3PubRecEncoder::new, Mqtt3PubRelEncoder.PROVIDER);
```

Comment:
Mir ist nicht ganz klar warum hier ein Mqtt3PubRelEncoder.PROVIDER mit übergeben wird. Was ist der Grund?

Reply:
Jede Nachricht kennt seinen EncoderProvider. Um das PubRel Antwortnachricht zu einer PubRec Nachricht erzeugen zu können, muss der PubRecEncoderProvider den zugehörigen PubRelEncoderProvider kennen.
Generell sind die Encoder so implementiert, dass Caching von errechneten Attributen möglich ist (z.B. property length in MQTT 5)